### PR TITLE
Improve serialization

### PR DESCRIPTION
**Pull request by cpcallen**
More tests (including a benchmark of post-roundtrip performance); fewer bugs.

### CODE REVIEW

File: server/tests/run.js [hunk 1]
```diff
@@ -175,5 +180,16 @@ function runTests() {
 
 /********************************************************************/
 
-runTests();
-runBenchmarks();
+/**
+ * Get list of test filenames matching a particular regexp.
+ * @param {!RegExp} pattern RegExp to match.
+ * @return {Array} List of filenames.
```
Comment (NeilFraser) [inline server/tests/run.js]: {!Array<string>} I think.

File: server/tests/serialize_bench.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        b = sum;
+      }
+    }
+    for(var i = 0; i < 10000; i++) {
+      var result = [];
+      fibonacci(78, result);
+    }
+    result;
+  `;
+  runInterpreterBench(b, name, src);
```
Comment (NeilFraser) [inline server/tests/serialize_bench.js]: b is a B or null, according to the JSDoc.  But runInterpreterBench only accepts a B according to its JSDoc.   Either add a `if (b)` check around this call, or require b be non-null in this function's JSDoc.

File: server/tests/serialize_test.js [hunk 3]
```diff
@@ -24,62 +24,124 @@
...
+ * - Run new instance to completion.
+ * - Append src2 (if specified) and run that to completion.
+ * - Verify result is as expected.
+ * @param {!T} t The test runner object.
+ * @param {string} name The name of the test.
+ * @param {string} src1 The code to be evaled before serialization.
+ * @param {string} src2 The code to be evaled after serialization.
+ * @param {number|string|boolean|null|undefined} expected The expected
+ *     completion value.
+ * @param {number=} stesp How many steps to run before serializing
```
Comment (NeilFraser) [inline server/tests/serialize_test.js]: stesp?

**Comment by cpcallen** [comment]
All fixed, I believe…

**Pull request merged**